### PR TITLE
Merging features necessary for Juju Min Version

### PR DESCRIPTION
**Pull request by kat-co**
This introduces a dependency on github.com/juju/version.

### CODE REVIEW

File: meta_test.go [hunk 1]
```diff
@@ -357,6 +358,43 @@ func (s *MetaSuite) TestInvalidSeries(c *gc.C) {
...
+
+// TestInvalidSeries ensures that invalid series values cause a parse error
+// when specified in the charm metadata.
+func (s *MetaSuite) TestInvalidMinJujuVersion(c *gc.C) {
+	_, err := charm.ReadMeta(strings.NewReader(dummyMetadata + "\nmin-juju-version: invalid-version"))
+
+	c.Check(err, gc.ErrorMatches, `invalid version "invalid-version"`)
+}
+
+// TestInvalidSeries ensures that invalid series values cause a parse error
```
Comment (wallyworld) [inline meta_test.go]: comment need fixing

File: meta_test.go [hunk 2]
```diff
@@ -357,6 +358,43 @@ func (s *MetaSuite) TestInvalidSeries(c *gc.C) {
...
+		c.Assert(meta.MinJujuVersion, gc.Equals, ver)
+	}
+}
+
+// TestInvalidSeries ensures that invalid series values cause a parse error
+// when specified in the charm metadata.
+func (s *MetaSuite) TestInvalidMinJujuVersion(c *gc.C) {
+	_, err := charm.ReadMeta(strings.NewReader(dummyMetadata + "\nmin-juju-version: invalid-version"))
+
+	c.Check(err, gc.ErrorMatches, `invalid version "invalid-version"`)
```
Comment (wallyworld) [inline meta_test.go]: can the error message be 'invalid min-version "foo"' so the user knows what attribute is wrong

**Comment by wallyworld** [comment]
LGTM with a couple of trivial fixes

**Comment by wallyworld** [comment]
If we stick with having the attr as a version.Number we'd add schema support for it. May a todo would be good

**Comment by natefinch** [comment]
LGTM

**Comment by kat-co** [comment]
$$merge$$

**Pull request merged**